disnake/poll.py

```diff
@@ -0,0 +1,426 @@
+            This attribute is not None only if you fetched the Poll object from the API.
+
+    allow_multiselect: :class:`bool`
+        Whether users are able to pick more than one answer.
+    layout_type: :class:`PollType`
+        The type of the layout of the poll.
+    results: Optional[:class:`PollResult`]
+        The results of the poll.
+
+        .. note::
+
+            This attribute is not None only if you fetched the Poll object from the API.
+    """
+
+    __slots__ = (
+        "_state",
+        "channel",
+        "message",
+        "question",
+        "answers",
+        "duration",
+        "expiry",
+        "allow_multiselect",
+        "layout_type",
+        "results",
+    )
+
+    def __init__(
+        self,
+        question: str,
+        *,
+        answers: List[PollAnswer],
+        duration: Optional[timedelta] = timedelta(hours=24),
+        allow_multiselect: bool = False,
+        layout_type: PollType = PollType.default,
+    ) -> None:
+        self._state: ConnectionState
+        self.channel: MessageableChannel
+        self.message: message.Message
+        self.expiry: datetime
+
+        self.question: str = question
+        self.answers: List[PollAnswer] = answers
+        self.results: Optional[PollResult] = None
+
+        self.duration: Optional[timedelta] = duration
+        if not duration:
+            self.duration = timedelta(hours=0)
+
+        self.allow_multiselect: bool = allow_multiselect
+        self.layout_type: PollType = layout_type
+
+    def __repr__(self) -> str:
+        return f"<{self.__class__.__name__} question={self.question} answers={self.answers!r}>"
+
+    @classmethod
+    def from_payload(
+        cls,
+        channel: MessageableChannel,
+        message: message.Message,
+        state: ConnectionState,
+        data: Optional[PollPayload] = None,
+    ) -> Optional[Poll]:
+        if data is None:
+            return None
+
+        poll = cls(
+            question=data["question"],
+            answers=[
+                PollAnswer.from_payload(state, channel.id, message.id, answer)
+                for answer in data["answers"]
+            ],
+            duration=None,
+            allow_multiselect=data["allow_multiselect"],
+            layout_type=try_enum(PollType, data["layout_type"]),
+        )
+        poll._state = state
+        poll.channel = channel
+        poll.message = message
+        poll.expiry = utils.parse_time(data["expiry"])
+        poll.results = None
+        if results := data["results"]:
+            poll.results = PollResult(results)
+
+        return poll
+
+    def _to_dict(self) -> PollCreatePayload:
+        # pyright is not able to tell that we are adding a keyword later,
+        # so we build answers first and then pass directly the list, building
+        # the dict object in one go
+        answers: List[PollCreateAnswerPayload] = []
+        for answer in self.answers:
+            answers.append(answer._to_dict())
```

Comment:
This can be inlined into the dict literal using `"answers": [answer._to_dict() for answer in self.answers]` c: